// @react-native-google-signin/google-signin/lib/commonjs/signIn/GoogleSignin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GoogleSignin = void 0;
var _reactNative = require("react-native");
var _NativeGoogleSignin = require("../spec/NativeGoogleSignin");
let configPromise = Promise.resolve();
function configure(options = {}) {
  if (options.offlineAccess && !options.webClientId) {
    throw new Error('RNGoogleSignin: offline use requires server web ClientID');
  }
  configPromise = _NativeGoogleSignin.NativeModule.configure(options);
}
async function signIn(options = {}) {
  await configPromise;
  return await _NativeGoogleSignin.NativeModule.signIn(options);
}
async function hasPlayServices(options = {
  showPlayServicesUpdateDialog: true
}) {
  if (_reactNative.Platform.OS === 'ios') {
    return true;
  } else {
    if (options && options.showPlayServicesUpdateDialog === undefined) {
      throw new Error('RNGoogleSignin: Missing property `showPlayServicesUpdateDialog` in options object for `hasPlayServices`');
    }
    return _NativeGoogleSignin.NativeModule.playServicesAvailable(options.showPlayServicesUpdateDialog);
  }
}
async function addScopes(options) {
  if (_reactNative.Platform.OS === 'ios') {
    return _NativeGoogleSignin.NativeModule.addScopes(options);
  } else {
    const hasUser = await _NativeGoogleSignin.NativeModule.addScopes(options);
    if (!hasUser) {
      return null;
    }
    // on Android, the user returned in onActivityResult() will contain only the scopes added, not the ones present previously
    // we work around it by calling signInSilently() which returns the user object with all scopes
    return signInSilently();
  }
}
async function signInSilently() {
  await configPromise;
  return _NativeGoogleSignin.NativeModule.signInSilently();
}
async function signOut() {
  return _NativeGoogleSignin.NativeModule.signOut();
}
async function revokeAccess() {
  return _NativeGoogleSignin.NativeModule.revokeAccess();
}
function hasPreviousSignIn() {
  return _NativeGoogleSignin.NativeModule.hasPreviousSignIn();
}
function getCurrentUser() {
  return _NativeGoogleSignin.NativeModule.getCurrentUser();
}
async function clearCachedAccessToken(tokenString) {
  if (!tokenString || typeof tokenString !== 'string') {
    return Promise.reject('GoogleSignIn: clearCachedAccessToken() expects a string token.');
  }
  return _reactNative.Platform.OS === 'ios' ? null : _NativeGoogleSignin.NativeModule.clearCachedAccessToken(tokenString);
}
async function getTokens() {
  if (_reactNative.Platform.OS === 'ios') {
    return _NativeGoogleSignin.NativeModule.getTokens();
  } else {
    const userObject = await _NativeGoogleSignin.NativeModule.getTokens();
    return {
      idToken: userObject.idToken,
      accessToken: userObject.accessToken
    };
  }
}

/**
 * The entry point of the Google Sign In API, exposed as `GoogleSignin`.
 * @group Original Google sign in
 * */
const GoogleSignin = exports.GoogleSignin = {
  hasPlayServices,
  configure,
  signIn,
  addScopes,
  signInSilently,
  signOut,
  revokeAccess,
  hasPreviousSignIn,
  getCurrentUser,
  clearCachedAccessToken,
  getTokens
};
